src/glang/parser/parser.c3

```diff
@@ -267,6 +282,38 @@ function get_next_decimal_token : (cursor : Cursor mut&) -> ProgramSlice = {
         if ascii_is_digit(next_char) {
             cursor.position += 1;
             token.end_offset += 1;
+
+            if cursor:get_char() == char(".") {
```

Comment:
This looks a bit weird. Is their a reason you've nested these inside `if ascii_is_digit(next_char)`? I suppose we don't have `else if` yet and you don't need to check if we've seen a digit already. 

What happens if `cursor:is_end_of_token() == true` and we call `cursor:get_char()` anyway?

Reply:
Yeah, I'm not really a fan of this tbh. We'd need better control flow to improve this ergonomically tho (`if` `else` at a minimum and ideally `break`/`continue`).

> What happens if cursor:is_end_of_token() == true and we call cursor:get_char() anyway?

This is fine, get_char() will just return whatever character caused the token to end (which here I'm fairly should be the "."). If `cursor:is_end_of_token() == true` and the next character is not the dot, we skip both branches and the loop ends normally.